### PR TITLE
Docker-compose with entrypoint

### PR DESCRIPTION
**Pull request by lakooi**
How to use:

- Define postgre password
- Run docker-compose up

### CODE REVIEW

File: server/Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+FROM node:16-alpine
+
+WORKDIR /app
+
+COPY . .
+
+RUN chown -R node:node ./*
+
+RUN npm install
```
Comment (no-comma) [inline server/Dockerfile.test:13]: Shouldn't this be `npm ci`

File: docker-tests/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  back-end:
+    build: 
+      context: ../server/
+      dockerfile: ../server/Dockerfile.test
+    depends_on:
+      - database
+    ports:
+    - "3000:3000"
+    container_name: backend-server
```
Comment (no-comma) [inline docker-tests/docker-compose.yaml:17]: I run into a naming conflict when I've also used the Docker Compose at the root of the project. This should probably be renamed.

File: server/Dockerfile.test [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+RUN npm install --save-dev sequelize-cli
```
Comment (no-comma) [inline server/Dockerfile.test:15]: Why are you installing `sequelize-cli` separately?

File: server/docker-run-tests.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# SPDX-FileCopyrightText: 2022 The Aalto Grades Developers
+#
+# SPDX-License-Identifier: MIT
+
+#!/bin/sh
+
+cat build/configs/database.js
```
Comment (no-comma) [inline server/docker-run-tests.sh:7]: This path is wrong, it should be `build/src/configs/database.js`. But why is this here?

**Review by no-comma — CHANGES_REQUESTED** [review]

**Comment by lakooi** [comment]
I noticed that the original docker-compose file for backend contained the same scripts as in my entrypoint.sh, so I am thus removing the custom dockerfile/bash file.

**Comment by no-comma** [comment]
Remember to update the year on new files to 2023.

File: docker-tests/docker-compose.yaml (hunk 2 above)
Comment (lakooi) [reply docker-tests/docker-compose.yaml:17]: Fixed

File: server/Dockerfile.test [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+RUN chown -R node:node ./*
+
+RUN npm ci
+
+RUN npm run build
+
+USER node
+
+CMD ["/bin/sh", "-c", "npm run migration:up && npm run seed:up && npm test"]
```
Comment (no-comma) [inline server/Dockerfile.test:19]: Is it possible to pass some argument to specify either test or start so we don't need two nearly identical Dockerfiles?

**Review by no-comma — COMMENTED** [review]
I think this is good now, but I won't approve it because it also contains changes from me.

**Review by Gskartwii — APPROVED** [review]
LGTM! Just curious, what's the purpose of the conditionality of `npm ci`?

**Comment by no-comma** [comment]
> LGTM! Just curious, what's the purpose of the conditionality of `npm ci`?

The main idea is to avoid unnecessarily running `npm ci` twice in the PR workflow in GitHub. Once outside the test container so the linter can be ran and then again while creating the test container.

**Review by jj-stigell — APPROVED** [review]
Looking good. Added fix to the test rerun issue by adding "seeds" table to the database that keeps track of seed. Sequelize will no skip loading seeds that have been loaded on previous test runs.
